### DISCUSSION TITLE
4 corner leveling is off. Nozzle is not above the adjusting knobs

### DISCUSSION THREAD

**Discussion by heidepiek** [The-300lb-Gorilla/Ender-2-Pro-v427#4] [Q&A] [answered]
Hello, i bought an ender 2 pro from 3djake For my son. He wanted to start 3d printing just like his dad. It was in sale so i decided to buy him one. But once purchased and installed we found out that the printer was not responding so wel with pronterface or octoprint.
So we searched the internet for the firmware of the ender 2 pro. I opend up the case of the pro and found out that it had the mcu hdsc board. Bummer. No source files. So we can`t tweak anything. Than i found this repo. I bought an v4.2.7 mainboard, solderd some cables.
And flashed it with your bin. This firmware of yours runs like clockwork!! Thanks for that. But a little remark. When using the 4 corner leveling,
when it moves from point to point, it is not directly on the wheel off the bed. Can we adjust this our selfs? Or do you have to do it in the firmware? Can you do this for us, or can we have the source file so that we can do it ourselfs.

Kind regards Harry and Mathijs
( ps Dutch, grammar may deviate)

![ender 2 mathijs](https://user-images.githubusercontent.com/104028471/227272022-941d8640-524d-488f-8e9e-f748298e2b27.jpeg)

![ender 2 pro 1](https://user-images.githubusercontent.com/104028471/227272077-b18d38a2-1ae5-4960-8405-e41e943c8556.jpeg)

![ender 2 pro 3](https://user-images.githubusercontent.com/104028471/227272179-418c690b-cddf-4dff-8b88-68cafde20431.jpeg)

**Comment by The-300lb-Gorilla** [comment]
I am so sorry for the long delay in answering back to you.
That version of the firmware is actually out of date. Great to see my work out in the wild for a change.
I actually just measured the offset my self with a metric ruler, and it is 30mmx30mm from each corner.
I checked the offsets in the Configuration.h file, and all are set to 30mm x 30mm on each corner.
However, the X Offset is different because of a custom print head I am using. My off sets are -7 on x and -4 on y. Factory is -18 on x and -6 on y. Let me change those to factory offsets, and I will post a new .bin file in a few minutes. Really sorry about that. I forgot to change them back after burning a .bin file for my all linear rail Ender 2 Pro. Give me until 8pm Eastern Standard Time, USA and it will be updated. You can see more on it here: https://www.printables.com/model/414535-ender-2-pro-x-y-z-linear-rail-conversion

**Comment by The-300lb-Gorilla** [comment, marked as answer]
Okay. I just uploaded it. It is in the https://github.com/The-300lb-Gorilla/Ender-2-Pro-v427  , the new .bin file is in the UPDATED FOR OFFSET FOR BED SCREWS Folder.
If you have any further issues, let me know. I built a new machine from scratch, so I do not really get a chance to use the Ender 2 Pro I have now. However, I do support my firmware designs, and I do provide follow up and support for my firmware designs.

**Comment by The-300lb-Gorilla** [comment]
I did just upload a newer version in the Ender 2 Pro v427 repository. Here
is a direct link:
https://github.com/The-300lb-Gorilla/Ender-2-Pro-v427/tree/main/Updated%20for%20Offset%20over%20Bed%20Screws
If anything else is needed, let me know. I am always happy to help. The
fastest way to reach me is on
https://www.printables.com/social/233133-the-300lb-gorilla/about
I check in there several times a day. Not so much on github, hence the
reason I took two days to reply.
Sincerely,
The 300lb Gorilla

**Reply by heidepiek** [reply]
No problem to wait a while , thanks for the new bin. Today I am not going to be able to flash the new firmware. My son is printing chess pieces. Won't bother him now
Tomorrow after work I will try the new bin.  Installed orange pi zero 2 last night on the ender 2 pro.  My 3d printers all have a raspberry pi with octoprint.  My son seemed to like that too. No dragging back and forth with the sd card.
But raspberry`s are expensive and almost impossible to get.  Then I read that obico sells a kit with an orange pi, must say that works well! Again a whole different way of installing. All abracadabra commands in putty. But it worked. Octoprint works well. But (host action commands ) is it enabled? With my other two printers I don`t get This message in octoprint 

( Your printer`s firmware supports host action commands, but they are disabled. The firmware will not inform OctoPrint about job pauses & cancellations, filament runouts, and similar firmware-side events. If you want this functionality, you need to change your firmware's configuration.)

On my other printers i use professional firmware. Also a nice firmware, til now no problems with printing.
Keep you posted on my findings of the new bin. Again thanks for the reply.

**Comment by heidepiek** [comment]
https://user-images.githubusercontent.com/104028471/227768917-a16dbfcc-2104-4330-9aab-228e1ab1f61d.mp4

**Comment by The-300lb-Gorilla** [comment]
Save your money on Raspberry pi or orange. Check out https://youtu.be/74xdib_-X38
[Thomas Sanladerer](https://www.youtube.com/@MadeWithLayers)
This is what I did on my custom 235 x235 x 275mm 3D Printer
I had a spare google phone, now it has a new purpose.

**Reply by heidepiek** [reply]
Thanks for the tip. 

Greetzz harrie

**Comment by heidepiek** [comment]
The new bin works!  There is stil a litle deviation. But the nozzle is close enough by the wheels. The printer also plays a tune when it starts.
So you have enabled speaker. Nice feature. Only one question left. could you define  HOST_ACTION_COMMANDS for us?


Greetzz from the netherlands!
Harrie and mathijs.

**Comment by The-300lb-Gorilla** [comment]
All set. Here is a direct link. Glad you like the music. That was kind of
an accident, but left it on to amuse users.
HOST ACTION COMMANDS .bin File Link:
https://github.com/The-300lb-Gorilla/Ender-2-Pro-v427/tree/main/Updated%20for%20Host%20Action%20Commands
Enjoy.
The 300lb Gorilla

**Comment by The-300lb-Gorilla** [comment]
Do me a favor, spread the word about the alternative v4.2.7 upgrade for the
Ender 2 Pro. Let people know they are not stuck with the NEW JUNK boards
the Ender 2 Pro is coming out with. That there is an alternative
option. You have been field testing the firmware on the go, and it looks
like it is working well. Share the link to my repository:
https://github.com/The-300lb-Gorilla/Ender-2-Pro-v427
Thank You in advance.
The 300lb Gorilla

**Reply by heidepiek** [reply]
If I get a chance to spread the word, I certainly will! But am not very active on social media myself. I am  a construction bench worker. Got into arduino a few years ago. Then sometimes needed something to be made by a 3d printer. Small store in our neighborhood with a 3d printer sometimes made something for me. But at a certain moment it became quite expensive. So I bought myself a 3d printer. Ender 3 v2. Good printer for beginners I was told. But every time I had problems with layer shifting. And got little feedback from creality. So persistent as I am, I started searching the world wide web for a solution. And then you find sites like github with people like you who have a solution. So I found out early on that creality sells nice products, but the help is nowhere to be found! Just like with the ender 2 pro, creality also messed around with the mainboards of the ender 3 v2.They threw an upgrade on the market, namely the v4.2.7. I had the v4.2.2, but as I said, through the smart people in the comunity I found out that there was a series of motherboards that had bad stepper drivers on them! And creality was definitely aware of this!!!  But let the vendors fix their mess! So got a new board from the seller. Not from creality!  But did offer an expensive upgrade, with supposedly silent drivers! Why not provide the printers they sell with the upgrade! Just money wolves! Now with the ender 2 pro this behavior repeats itself again. And if they just released the source now! So they don't do that either. So again I had to rely on the world wide web.  Bought the software from th3d studios. 5 euros.
And wanted to buy the cables also, but including shipping to netherlands 56 euro. Hmm, that was too much for a few cables. So ordered a set of plugs from tinytronics here in the Netherlands, and made some cables myself. But I also found your repository, so I first flashed your firmware on the printer. And I like it! Maybe I will try the one from th3d studios as well. But not yet. 4 button bed leveling, and manual mesh. Octoprint works well now, and music. What more could you want. Happy printing mister gorilla, and thanks again for the fine tuned firmware. Glad that there are people willing to help each other!

Greetzz harrie
![ender 2 1](https://user-images.githubusercontent.com/104028471/228609331-af5425e3-d1d5-4690-9bfc-58ea78e8f13e.jpeg)
![ender 2 2](https://user-images.githubusercontent.com/104028471/228609362-b06c3bdf-dc09-432d-8d19-99a33769958f.jpeg)
![ender 2 3](https://user-images.githubusercontent.com/104028471/228609397-96884b0a-6e14-4fe3-951f-10985dc5c231.jpeg)
![ender 2 4](https://user-images.githubusercontent.com/104028471/228609417-f8c89e46-b500-4c6f-baa7-ef9790d951d1.jpeg)
![ender 2 5](https://user-images.githubusercontent.com/104028471/228609430-09845527-0e01-49fa-88b2-fcbdf678a0c4.jpeg)

https://user-images.githubusercontent.com/104028471/228612642-ed96976a-a253-490e-ae47-d259575570ce.mp4

In the video the printer plays a tune before printing.

**Comment by The-300lb-Gorilla** [comment]
The TH3DStudio firmware is a waste of money. Found that out the hard way before I got back into programming. Glad to hear the Ender 2 Pro 4.2.7B firmware is working for you. Should you ever need anything else, feel free to drop me a note in here, on PRINTABLES.COM
I use the same name there, so it will not be hard to find me.

**Reply by heidepiek** [reply]
Hello mr gorilla, no longer member of printables.com? 

Greetzz harrrie

**Reply by The-300lb-Gorilla** [reply]
No. I am no longer a member of printables. To much whining and snitching, and back stabbing. I followed the rules on posting other people's models to the letter! Still got threats, was snitched on for nothing. I believe it was mostly because I had thousands and thousands of downloads, and I was a Level 25, and people were jealous. I went back to thingiver.com. May be clunky, may be slow to update, but at least people over there leave me the hell alone. You can find me at: https://www.thingiverse.com/the-3oolb-gorilla/designs
